flink-python/pyflink/datastream/connectors/kinesis.py

```diff
@@ -0,0 +1,368 @@
+
+    - maxBatchSize will be 500
+    - maxInFlightRequests will be 50
+    - maxBufferedRequests will be 10000
+    - maxBatchSizeInBytes will be 5 MB i.e. 5 * 1024 * 1024
+    - maxTimeInBufferMS will be 5000ms
+    - maxRecordSizeInBytes will be 1 MB i.e. 1 * 1024 * 1024
+    - failOnError will be false
+    """
+
+    def __init__(self):
+        super().__init__()
+        JKinesisSink = get_gateway().jvm.org.apache.flink.connector.kinesis.sink.KinesisStreamsSink
+        self._j_kinesis_sink_builder = JKinesisSink.builder()
+
+    def set_stream_name(self, stream_name: Union[str, List[str]]) -> 'KinesisSinkBuilder':
+        """
+        Sets the name of the KDS stream that the sink will connect to. There is no default for this
+        parameter, therefore, this must be provided at sink creation time otherwise the build will
+        fail.
+        """
+        self._j_kinesis_sink_builder.setStreamName(stream_name)
+        return self
+
+    def set_serialization_schema(self, serialization_schema: SerializationSchema) \
+            -> 'KinesisSinkBuilder':
+        """
+        Sets the SerializationSchema of the KinesisSinkBuilder.
+        """
+        self._j_kinesis_sink_builder.setSerializationSchema(
+            serialization_schema._j_serialization_schema)
+        return self
+
+    def set_partition_key_generator(self, partition_key_generator: PartitionKeyGenerator) \
+            -> 'KinesisSinkBuilder':
+        """
+        Sets the PartitionKeyGenerator of the KinesisSinkBuilder.
+        """
+        self._j_kinesis_sink_builder.setPartitionKeyGenerator(
+            partition_key_generator.j_partition_key_generator)
+        return self
+
+    def set_fail_on_error(self, fail_on_error: bool) -> 'KinesisSinkBuilder':
+        """
+        Sets the failOnError of the KinesisSinkBuilder. If failOnError is on, then a runtime
+        exception will be raised. Otherwise, those records will be requested in the buffer for
+        retry.
+        """
+        self._j_kinesis_sink_builder.setFailOnError(fail_on_error)
+        return self
+
+    def set_kinesis_client_properties(self, kinesis_client_properties: Dict) \
+            -> 'KinesisSinkBuilder':
+        """
+        Sets the kinesisClientProperties of the KinesisSinkBuilder.
+        """
+        j_properties = get_gateway().jvm.java.util.Properties()
+        for key, value in kinesis_client_properties.items():
+            j_properties.setProperty(key, value)
+        self._j_kinesis_sink_builder.setKinesisClientProperties(j_properties)
+        return self
+
+    def set_max_batch_size(self, max_batch_size: int) -> 'KinesisSinkBuilder':
+        """
+        Maximum number of elements that may be passed in a list to be written downstream.
+        """
+        self._j_kinesis_sink_builder.setMaxBatchSize(max_batch_size)
+        return self
+
+    def set_max_in_flight_requests(self, max_in_flight_requests):
```

Comment:
```suggestion
    def set_max_in_flight_requests(self, max_in_flight_requests: int):
```

Also check the below methods.